docs/10-architecture.md

```diff
@@ -0,0 +1,130 @@
+    classDef Stream stroke-dasharray: 5 5
+```
+
+## Design
+
+Media Atom Maker (MAM) is architected as a collection of resources, primarily a Scala Play App with a React frontend
+that facilitates the management and publishing of video content, typically hosted in YouTube. This system is built using
+various AWS services, including S3 buckets, Lambdas, Step Functions, and DynamoDB for content state management. It
+integrates with Composer, YouTube, the Content API (CAPI), and Pluto for content its publishing workflows.
+
+### EC2 Backend
+
+The main backend of MAM is a Scala Play App running on EC2 serving the client side and providing an API to manage media
+atoms and other metadata. It manages the content state using DynamoDB tables.
+
+### Client Side
+
+The client side is a React app bundled using Vite and served via the Play app. The project uses Redux for state
+management.
+
+### Uploader
+
+The backend grants browser credentials for direct upload of media content to an S3 bucket. Objects are then picked up by
+the Uploader step function and processed. These assets are either sent to YouTube or processed and added to the
+`uploads.guim.co.uk` S3 bucket.
+
+[Uploader Step Function README](../uploader/README.md)
+
+### Cloudformation Stacks
+
+The application has several cloudformation stacks:
+
+- media-atom-maker-{Stage} – Primary resources stack
+- media-atom-maker-{Stage}-dynamo – DynamoDB tables (AtomMaker, PublishedAtomMaker and ManualPlutoAtomMaker)
+- media-atom-pipeline-{Stage} – Uploader step function resources
+- media-atom-maker-DEV – Resource for integration with local testing (different template to the primary stack)
+
+## Integrations
+
+### Pluto
+
+Pluto is the Guardian's media asset management system, confusingly also referred to as MAM. Its taxonomy consists of
+Commissions, Projects and Deliverables. It provides a way to manage media assets, and expects users to have access to
+the shared asset drive.
+
+Media Atom Maker and Pluto communicate via Kinesis and SQS to transmit updates and metadata between each system.
+
+Commissions and Projects are ingested via the Pluto Message Ingestion Kinesis stream and stored in DynamoDB. These can
+then be applied to media atoms to provide navigation between content and their assets.
+
+Media Atom Maker events are sent to Pluto via the Upload To Pluto Kinesis stream. These include notifications of when
+videos are uploaded, and when Pluto projects are assigned to atoms
+
+[Pluto Message Ingestion](../pluto-message-ingestion/README.md)
+
+### CAPI
+
+Media atoms are published via CAPI using the Publish Atom SNS topic.
+
+The Expirer lambda periodically checks the state of media atoms in CAPI and if they are expired updates their status in
+YouTube to private.
+
+The Scheduler lambda periodically checks the state of media atoms in CAPI and if they are scheduled for launch sends a
+request to the media atom maker backend API to publish them.
```

Comment:
I was confused by this at first because I didn’t understand why media atoms would already be in CAPI before they’ve launched. Looking at the code it looks like this is querying preview CAPI, which explains that – is it worth explicitly saying the lambda is querying preview CAPI here, or is that unnecessary complication?